applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_broadcast.c

```diff
@@ -256,6 +256,142 @@ static struct bt_le_per_adv_sync_cb sync_callbacks = {
 	.term = pa_sync_terminated_cb,
 };
 
+static void pa_timer_handler(struct k_work *work)
+{
+	int ret;
+
+	if (req_recv_state != NULL) {
+		enum bt_bap_pa_state pa_state;
+
+		if (req_recv_state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ) {
+			pa_state = BT_BAP_PA_STATE_NO_PAST;
+		} else {
+			pa_state = BT_BAP_PA_STATE_FAILED;
+		}
+
+		ret = bt_bap_scan_delegator_set_pa_state(req_recv_state->src_id, pa_state);
+		if (ret) {
+			LOG_ERR("set PA state to %d failed, err = %d", pa_state, ret);
+		}
+	}
+}
+
+static K_WORK_DELAYABLE_DEFINE(pa_timer, pa_timer_handler);
+
+static int pa_sync_past(struct bt_conn *conn, uint16_t pa_interval)
+{
+	int ret;
+	struct bt_le_per_adv_sync_transfer_param param = {0};
+
+	param.skip = PA_SYNC_SKIP;
+	param.timeout = interval_to_sync_timeout(pa_interval);
+
+	ret = bt_le_per_adv_sync_transfer_subscribe(conn, &param);
+	if (ret) {
+		LOG_WRN("Could not do PAST subscribe: %d", ret);
+		return ret;
+	}
+
+	LOG_DBG("Syncing with PAST: %d", ret);
+	(void)k_work_reschedule(&pa_timer, K_MSEC(param.timeout * 10));
+
+	return ret;
+}
+
+static int pa_sync_req_cb(struct bt_conn *conn,
+			  const struct bt_bap_scan_delegator_recv_state *recv_state,
+			  bool past_avail, uint16_t pa_interval)
+{
+	int ret;
+
+	req_recv_state = recv_state;
+
+	if (recv_state->pa_sync_state == BT_BAP_PA_STATE_SYNCED ||
+	    recv_state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ) {
+		LOG_DBG("Already syncing");
+		/* TODO: Terminate existing sync and then sync to new?*/
+		return -EALREADY;
+	}
+
+	LOG_INF("broadcast ID received = %X", recv_state->broadcast_id);
+	brcast_src_info.id = recv_state->broadcast_id;
+
+	if (IS_ENABLED(CONFIG_BT_PER_ADV_SYNC_TRANSFER_RECEIVER) && past_avail) {
+		ret = pa_sync_past(conn, pa_interval);
+		if (ret) {
+			LOG_WRN("Subscribe to PA sync PAST failed, ret = %d", ret);
+			return ret;
+		}
+
+		ret = bt_bap_scan_delegator_set_pa_state(req_recv_state->src_id,
+							 BT_BAP_PA_STATE_INFO_REQ);
+		if (ret) {
+			LOG_WRN("Set PA state to INFO_REQ failed, err = %d", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int pa_sync_term_req_cb(struct bt_conn *conn,
+			       const struct bt_bap_scan_delegator_recv_state *recv_state)
+{
+	int ret;
+	struct bt_mgmt_msg msg;
+
+	msg.event = BT_MGMT_BROADCAST_SINK_DISABLE;
+	ret = zbus_chan_pub(&bt_mgmt_chan, &msg, K_NO_WAIT);
+	ERR_CHK(ret);
```

Comment:
Since this returns an int, should we use ERR_CHK?

Reply:
No, we should return ret here, I believe that will notify the mobile phone that the termination request failed

Reply:
This is not changed as I can see.

Reply:
I will trigger an error for testing and back to here.

Reply:
What was the result of the testing?

Reply:
I tried to return -1, but the broadcast assistant behavior doesn't change on the phone side.
To me, I would suggest we remain ERR_CHK at this moment. Since we should not fail on publishing events.